src/app/courses/courses-list/courses-list.component.ts

```diff
@@ -1,62 +1,102 @@
-import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Component, OnInit, Input, OnChanges, OnDestroy } from '@angular/core';
+import { debounceTime } from 'rxjs';
 
-import { Course } from 'src/app/shared/interfaces/course.interface.';
-import { FilterPipe } from 'src/app/shared/pipes/filter.pipe';
+import { HttpCourse } from 'src/app/shared/interfaces/http-course.interface';
 import { CourseService } from 'src/app/shared/services/course.service';
 
 @Component({
   selector: 'app-courses-list',
   templateUrl: './courses-list.component.html',
   styleUrls: ['./courses-list.component.css']
 })
-export class CoursesListComponent implements OnInit {
-
-  constructor(private coursesService: CourseService) {}
+export class CoursesListComponent implements OnInit, OnChanges {
 
   @Input() searchValue: string = '';
-
-  courses: Course[] = [];
-  courseForDelete?: Course;
-  titleForDelete = "";
+  courseForDelete?: HttpCourse;
+  titleForDelete: string = '';
   showModal = false;
+  listOfCourses: HttpCourse[] = [];
+  inputValue = '';
+  debounceTimeValue = 500;
 
-
-  ngOnInit() {
-      this.courses = this.coursesService.getListCourses();
+  constructor(private courseService: CourseService ) {
+    
   }
-
-  trackByFn(index: number, item: Course): number {
-    return item.id; // Assuming each item has a unique "id" property
+  
+  ngOnInit() {
+    this.courseService.loadingSubject.next(true);
+    this.courseService.onFetchCourses()
+    .subscribe(
+      courses => {
+        this.listOfCourses = [...courses];
```

Comment:
Do you need to create a new array here?

Reply:
I just didn't want to mutate the object

Reply:
Usually you get such object from the backend, so no need to create the new array